Spigot-Server-Patches/0336-Extend-player-profile-API-to-support-skin-changes.patch

```diff
@@ -0,0 +1,191 @@
+From e709f6448451449a488b16d4039b25ff467cc956 Mon Sep 17 00:00:00 2001
+From: NickAcPT <32451103+NickAcPT@users.noreply.github.com>
+Date: Sat, 21 Jul 2018 01:30:30 +0100
+Subject: [PATCH] Extend player profile API to support skin changes
+
+Added code that refreshes the player's skin by sending packets with a special order, telling the client to respawn the player and re-apply the game profile
+
+diff --git a/src/main/java/com/destroystokyo/paper/profile/CraftPlayerProfile.java b/src/main/java/com/destroystokyo/paper/profile/CraftPlayerProfile.java
+index 9ad5853d..dda24052 100644
+--- a/src/main/java/com/destroystokyo/paper/profile/CraftPlayerProfile.java
++++ b/src/main/java/com/destroystokyo/paper/profile/CraftPlayerProfile.java
+@@ -170,17 +170,21 @@ public class CraftPlayerProfile implements PlayerProfile {
+     }
+ 
+     public boolean complete(boolean textures) {
++        return complete(textures, false);
++    }
++
++    public boolean complete(boolean textures, boolean force) {
+         MinecraftServer server = MinecraftServer.getServer();
+ 
+         boolean isOnlineMode = server.getOnlineMode() || (SpigotConfig.bungee && PaperConfig.bungeeOnlineMode);
+         boolean isCompleteFromCache = this.completeFromCache(true);
+-        if (isOnlineMode && (!isCompleteFromCache || textures && !hasTextures())) {
```

Comment:
See API review, this is unneeded. isOnlineMode is true if the server is on offline mode but has the paper online bungee mode turned on.